src/azure-cli/azure/cli/command_modules/synapse/manual/_client_factory.py

```diff
@@ -108,10 +108,7 @@ def synapse_spark_factory(cli_ctx, workspace_name, sparkpool_name):
     from azure.cli.core.commands.client_factory import get_subscription_id
     subscription_id = get_subscription_id(cli_ctx)
     profile = Profile(cli_ctx=cli_ctx)
-    cred, _, _ = profile.get_login_credentials(
-        resource=cli_ctx.cloud.endpoints.synapse_analytics_resource_id,
-        subscription_id=subscription_id
-    )
+    cred, _, _ = profile.get_login_credentials(subscription_id=subscription_id)
```

Comment:
Please double check if `cli_ctx.cloud.endpoints.synapse_analytics_resource_id` needs to be set as `credential_scopes` when creating `SparkClient`, `AccessControlClient`, `ArtifactsClient` and `VnetClient`.